server/src/main/java/io/crate/analyze/Analyzer.java

```diff
@@ -188,6 +191,10 @@ public Analyzer(Schemas schemas,
         this.resetStatementAnalyzer = new ResetStatementAnalyzer(functions);
     }
 
+    public NodeContext nodeContext() {
```

Comment:
Would be cool if we could avoid this. Using a class as both (not encapsulated) data carrier and as a component that provides some other fucntionality is a sign that it breaks separation of concerns.

I get that this may be convenient and "easier" - as less changes necessary and less to type, but it may make changes more difficult down the road.